src/App.tsx

```diff
@@ -14,11 +17,141 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[];
+  isListVisible: boolean;
+  isReversed: boolean;
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [...goodsFromServer],
+    isListVisible: false,
+    isReversed: false,
+  };
+
+  showList = () => {
+    this.setState({ isListVisible: true });
+  };
+
+  reverseList = () => {
+    const { isListVisible } = this.state;
+
+    if (isListVisible) {
+      this.setState(state => ({
+        isReversed: !state.isReversed,
+      }));
+    }
+  };
+
+  reset = () => {
+    this.setState(state => ({
+      goods: [...goodsFromServer],
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByAlphabet = () => {
+    this.setState(state => ({
+      goods: [...state.goods.sort((a, b) => b.localeCompare(a))],
```

Comment:
you can join the functions using switch construction for different types of properties